File: Bars.GkhGJI/Entities/MKDLicRequest/MKDLicRequestFile.cs

namespace Bars.GkhGji.Entities
{
    using Bars.B4.DataAccess;
    using Bars.B4.Modules.FileStorage;
    using Enums;
    using System;

    /// <summary>
    /// Прилагаемые документы
    /// </summary>
    public class MKDLicRequestFile : BaseEntity
    {
        /// <summary>
        /// MKDLicRequest
        /// </summary>
        public virtual MKDLicRequest MKDLicRequest { get; set; }

        /// <summary>
        ///Файл
        /// </summary>
        public virtual  FileInfo FileInfo { get; set; }

        /// <summary>
        /// Тип документа
        /// </summary>
        public virtual LicStatementDocType LicStatementDocType { get; set; }

        /// <summary>
        /// Наименование документа
        /// </summary>
        public virtual string DocumentName { get; set; }

        /// <summary>
        /// Комментарий
        /// </summary>
        public virtual string Description { get; set; }

        /// <summary>
        /// Дата документа
        /// </summary>
        public virtual DateTime? DocDate { get; set; }

        /// <summary>
        /// Подписанный файл
        /// </summary>
        public virtual FileInfo SignedFile { get; set; }

        /// <summary>
        /// Подпись
        /// </summary>
        public virtual FileInfo Signature { get; set; }

        /// <summary>
        /// Сертификат
        /// </summary>
        public virtual FileInfo Certificate { get; set; }
    }
}

File: Bars.GkhGJI/libs/B4/model/mkdlicrequest/MKDLicRequestFile.js
Ext.define('B4.model.mkdlicrequest.MKDLicRequestFile', {
    extend: 'B4.base.Model',
    idProperty: 'Id',

    proxy: {
        type: 'b4proxy',
        controllerName: 'MKDLicRequestFile'
    },
    fields: [
        { name: 'Id' },
        { name: 'MKDLicRequest'},
        { name: 'FileInfo'},
        { name: 'SignedFile'},
        { name: 'DocumentName' },
        { name: 'LicStatementDocType', defaultValue: 0},
        { name: 'Description' },
        { name: 'DocDate' }
       
    ]
});